### PR TITLE
CE-224: update dockerfile

### PR DESCRIPTION
**Pull request by ialarmedalien**
Updating the docker file to a much slimmer version without kb-sdk, and adding the compile report to the repo instead.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 test_local
 sdk.cfg
+Dockerfile
+Makefile
```
Comment (ialarmedalien) [inline .dockerignore:4]: we won't be running kb-sdk on the image, so no need for the makefile

File: Makefile [hunk 2]
```diff
@@ -19,7 +20,9 @@ default: compile
 all: compile build set-executable
 
 compile:
-	kb-sdk compile $(SPEC_FILE) \
+	rm $(COMPILE_REPORT)
+	KB_SDK_COMPILE_REPORT_FILE=$(COMPILE_REPORT) kb-sdk compile $(SPEC_FILE) \
```
Comment (ialarmedalien) [inline Makefile:24]: ensure the compile report gets generated

File: README.md [hunk 3]
```diff
@@ -55,7 +55,7 @@ $ pytest test
 
 ## API changes
 
-If you make changes to the spec (and thus the API) of the StaticNarrative app, you will need to recompile the app to generate updated versions of `StaticNarrativeImpl.py` and `StaticNarrativeServer.py`.
+If you make changes to the spec (and thus the API) of the StaticNarrative app, you will need to recompile the app to generate updated versions of `StaticNarrativeImpl.py`, `StaticNarrativeServer.py`, and the compilation report, `compile_report.json`. These updated files must be committed to the GitHub repo.
```
Comment (bio-boris) [inline README.md:58]: Maybe this should include a pull request checklist to ensure the developers do these steps!!

Comment (ialarmedalien) [reply README.md:58]: Good idea, thanks!

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+# Description of PR purpose/changes
```
Comment (ialarmedalien) [inline .github/pull_request_template.md:1]: Generic PR template... you may recognise it!

File: scripts/entrypoint.sh [hunk 5]
```diff
@@ -12,16 +10,16 @@ if [ $# -eq 0 ] ; then
...
 elif [ "${1}" = "bash" ] ; then
-  bash
+  echo "This image only supports sh shells"
+  sh
+elif [ "${1}" = "sh" ] ; then
+  sh
 elif [ "${1}" = "report" ] ; then
-  export KB_SDK_COMPILE_REPORT_FILE=./work/compile_report.json
-  make compile
+  echo "The compile report can be found at ./work/compile_report.json"
```
Comment (ialarmedalien) [inline scripts/entrypoint.sh:22]: remove the compilation step as it's only used to generate the compile report file, which is already present in the repo.

File: Dockerfile [hunk 6]
```diff
@@ -1,34 +1,19 @@
...
 
+COPY ./ /kb/module/
 WORKDIR /kb/module
+# install python modules one at a time so that all deps get resolved properly
+RUN pip install --upgrade pip && \
+    cat requirements.txt | sed -e '/^\s*#.*$/d' -e '/^\s*$/d' | xargs -n 1 pip install && \
+    cat requirements-test.txt | sed -e '/^\s*#.*$/d' -e '/^\s*$/d' | xargs -n 1 pip install && \
+    mkdir -p /kb/module/work && \
+    chmod -R a+rw /kb/module && \
+    cp compile_report.json work/
```
Comment (ialarmedalien) [inline Dockerfile:12]: copy the compile_report over to the work directory, which is where the Catalog expects it to be

File: Dockerfile [hunk 7]
```diff
@@ -1,34 +1,19 @@
...
-    requests-mock==1.7.0 \
-    jsonrpcbase==0.2.0 \
-    python-dateutil==2.8.1
-
-# -----------------------------------------
-
-COPY ./ /kb/module
-RUN mkdir -p /kb/module/work
-RUN chmod -R a+rw /kb/module
+FROM python:3.10-alpine
```
Comment (briehl) [inline Dockerfile:1]: 🎉

**Pull request merged**